components/sidebar/sidebar_category/sidebar_category_sorting_menu/sidebar_category_sorting_menu.tsx

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+
+import {ChannelCategory, CategorySorting} from 'mattermost-redux/types/channel_categories';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
+
+import Constants from 'utils/constants';
+
+import {trackEvent} from 'actions/telemetry_actions';
+
+import SidebarMenu from 'components/sidebar/sidebar_menu';
+import SidebarMenuType from 'components/sidebar/sidebar_menu/sidebar_menu';
+import Menu from 'components/widgets/menu/menu';
+import {Props as SubmenuItemProps} from 'components/widgets/menu/menu_items/submenu_item';
+
+type Props = {
+    category: ChannelCategory;
+    handleOpenMoreDirectChannelsModal: (e: Event) => void;
+    categories?: ChannelCategory[];
+    intl: IntlShape;
+    isCollapsed: boolean;
+    isMenuOpen: boolean;
+    onToggleMenu: (isMenuOpen: boolean) => void;
+    currentUserId: string;
+    actions: {
+        setCategorySorting: (categoryId: string, sorting: CategorySorting) => void;
+        savePreferences: (userId: string, preferences: PreferenceType[]) => void;
+    };
+};
+
+type State = {
+    openUp: boolean;
+    width: number;
+    selectedDmNumber: number;
+    selectedCategorySort: string;
+};
+
+enum Settings {
+    limitVisibleDmsGms = 'limit_visible_dms_gms',
+}
+
+export class SidebarCategorySortingMenu extends React.PureComponent<Props, State> {
+    isLeaving: boolean;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            openUp: false,
+            width: 0,
+            selectedDmNumber: 20,
+            selectedCategorySort: CategorySorting.Alphabetical,
+        };
+
+        this.isLeaving = false;
+    }
+
+    handleSortDirectMessages = (sorting: CategorySorting) => {
+        const {category} = this.props;
+
+        this.props.actions.setCategorySorting(category.id, sorting);
+        trackEvent('ui', `ui_sidebar_sort_dm_${sorting}`);
```

Comment:
@esethna This makes me think, do we want telemetry events for this?

Reply:
I mean, do we want telemetry events when a user changes their number of visible DMs?